creator-node/src/fileManager.js

```diff
@@ -352,72 +414,101 @@ async function saveFileForMultihashToFS(
+      (c) =>
+        !targetGateways.includes(c) && config.get('creatorNodeEndpoint') !== c
+    )
 
-      return baseUrl
-    })
+    // if this is a directory, make it compatible with our dir cid gateway url
+    // [https://endpoint.co/ipfs/Qm111/150x150.jpg, https://endpoint.co/ipfs/Qm222/150x150.jpg ...]
+    if (dirCID && fileNameForImage) {
+      targetGatewayContentRoutes = targetGateways.map(
+        (endpoint) =>
+          `${endpoint.replace(/\/$/, '')}/ipfs/${dirCID}/${fileNameForImage}`
+      )
+      nonTargetGatewayContentRoutes = nonTargetGateways.map(
+        (endpoint) =>
+          `${endpoint.replace(/\/$/, '')}/ipfs/${dirCID}/${fileNameForImage}`
+      )
+      decisionTree.recordStage({
+        name: 'Set targetGatewayContentRoutes and nonTargetGatewayContentRoutes for dir',
+        data: { nonTargetGatewayContentRoutes, targetGatewayContentRoutes }
+      })
+
+      storageLocation = await DiskManager.computeFilePathInDirAndEnsureItExists(
+        dirCID,
+        multihash
+      )
+    } else {
+      // if it's not a directory, make it a regular gateway route
+      // eg [https://endpoint.co/ipfs/Qm111, https://endpoint.co/ipfs/Qm222 ...]
+      // TODO - don't concat url's by hand like this, use module like urljoin
+      // ..replace(/\/$/, "") removes trailing slashes
+
+      targetGatewayContentRoutes = targetGateways.map((endpoint) => {
+        let baseUrl = `${endpoint.replace(/\/$/, '')}/ipfs/${multihash}`
+        if (trackId) baseUrl += `?trackId=${trackId}`
+
+        return baseUrl
+      })
 
-    const parsedStoragePath = path.parse(expectedStoragePath).dir
+      nonTargetGatewayContentRoutes = nonTargetGateways.map((endpoint) => {
+        let baseUrl = `${endpoint.replace(/\/$/, '')}/ipfs/${multihash}`
+        if (trackId) baseUrl += `?trackId=${trackId}`
+
+        return baseUrl
+      })
+
+      storageLocation = await DiskManager.computeFilePathAndEnsureItExists(
+        multihash
+      )
+
+      decisionTree.recordStage({
+        name: 'Set targetGatewayContentRoutes and nonTargetGatewayContentRoutes for non-dir file',
+        data: { nonTargetGatewayContentRoutes, targetGatewayContentRoutes }
+      })
+    }
+
+    const storageLocationParentDir = path.parse(storageLocation).dir
 
     decisionTree.recordStage({
       name: 'About to start running saveFileForMultihashToFS()',
       data: {
         multihash,
-        expectedStoragePath,
-        parsedStoragePath
+        storageLocation,
+        storageLocationParentDir
       },
-      log: true,
       logLevel: 'debug'
     })
 
     // Create dir at expected storage path in which to store retrieved data
     try {
       // calling this on an existing directory doesn't overwrite the existing data or throw an error
       // the mkdir recursive is equivalent to `mkdir -p`
-      await fs.mkdir(parsedStoragePath, { recursive: true })
+      await fs.mkdir(storageLocationParentDir, { recursive: true })
```

Comment:
isn't this redundant now since the above computePath calls already ensureItExists?